packages/core/src/lib/input/slider-input.svelte

```diff
@@ -0,0 +1,157 @@
+<!--
+@component
+  
+For numeric user inputs that require easy adjustment.
+
+```svelte
+<SliderInput min={0} max={1} step={0.1}  on:input={onInput} />
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import { createEventDispatcher, tick } from 'svelte';
+import Tooltip from '$lib/tooltip.svelte';
+import NumericInput from './numeric-input.svelte';
+import {
+  getDecimals,
+  parseNumericInputValue,
+  type NumericInputTypes,
+} from './utils';
+
+/** The input type */
+export let type: NumericInputTypes = 'number';
+
+export let value: number | undefined = undefined;
+
+export let step = 1;
+
+export let min = -Infinity;
+export let max = Infinity;
+
+const dispatch = createEventDispatcher<{
+  input: number;
+  blur: void;
+  keydown: number;
+}>();
+
+let numberDragTooltip: Tooltip;
+let numberDragCord: HTMLDivElement;
+let numberDragHead: HTMLDivElement;
+let isDragging = false;
+let startX = 0;
+let startValue = 0;
+
+let stepDecimalDigits = 0;
+$: {
+  const decimal = step % 1;
+  stepDecimalDigits = decimal === 0 ? 0 : `${decimal}`.length;
+}
+
+$: isNumber = type === 'number';
+$: parse = (value: string) => parseNumericInputValue(value, type);
+
+const handleNumberDragMove = (event: PointerEvent) => {
+  const x = event.clientX;
+  const deltaString = ((-(startX - x) * step) / 10).toFixed(
+    isNumber ? stepDecimalDigits : 0
+  );
+
+  const inputValue = (event.target as HTMLInputElement).value;
+  const delta = parse(deltaString);
+  const next = parse(
+    (startValue + delta * step).toFixed(getDecimals(inputValue))
+  );
+
+  if (next > max) {
+    value = max;
+    return;
+  }
+
+  if (next < min) {
+    value = min;
+    return;
+  }
+
+  if (next > startValue) {
+    const dx = x - startX;
+    numberDragHead.style.transform = `translate(${dx}px, 0px)`;
+    numberDragCord.style.cssText = `width: ${dx}px;`;
+  } else if (next < startValue) {
+    const dx = startX - x;
+    numberDragHead.style.transform = `translate(-${dx}px, 0px)`;
+    numberDragCord.style.cssText = `
+      width: ${dx}px;
+      transform: translate(-${dx}px, 0);
+    `;
+  }
+
+  value = next;
+  dispatch('input', value);
+
+  // The tooltip may not be mounted the first time this handler is called.
+  numberDragTooltip.recalculateStyle();
+};
+
+const handleNumberDragUp = () => {
+  isDragging = false;
+  window.removeEventListener('pointermove', handleNumberDragMove);
```

Comment:
Looks like the pointerup listener is not removed

Reply:
It is set to only fire once, not sure if it is necessary to remove it: https://github.com/viamrobotics/prime/pull/320/files#diff-159329097f01d3404df5ec038d4ca2a7beeac4648d1595cdbe32f70860351958R118